src/Application/ApplicationFileProcessor.php

```diff
@@ -94,9 +100,16 @@ public function run(Configuration $configuration, InputInterface $input): Proces
         }
 
         if ($configuration->isParallel()) {
+            $this->fileDependenciesCache->cacheAllFiles($allFiles);
```

Comment:
If we are running in parallel, we save the list of all possible files in the cache. We do this because calculating this list is an expensive operation and calculating it for each worker would slow the process. Not sure this is the best option to pass this array to the workers, if you can think of a better option, please let me know